build/docker-bootstrap.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+# `make` for ruby-install and chruby don't work within our strict environment
+# so we have to unset the BASH_ENV variable before executing their setup.
+# The `make` call creates a new shell so doing `set +o [option]` doesn't impact
+# the `make install` execution
+unset_strict(){
+  unset BASH_ENV
+}
+
+# Change directories to either chruby's or ruby-install's source directory
+# and `make install`
+install_manager(){
+  local manager_dir="$1"
+
+  (
+    unset_strict;
+    cd "$manager_dir" || exit 1;
+    make install
+  )
+}
+
+# Install bundler for a given ruby version
+install_bundler_for_ruby_version(){
+  local ruby_version="$1"
+
+  # chruby unsets variables and uses file globbing
+  # execute in a subshell so that environment changes for chruby don't have
+  # global effect
+  (
+    # Only unset the rules that are required to not break chruby
+    set +o nounset;
+    set +o noglob;
+    source "/usr/local/share/chruby/chruby.sh"
+    chruby "$ruby_version"
+    gem install bundler
+  )
+}
+
+# Use ruby-install to install a ruby version
+install_ruby_version(){
+  local ruby_version="$1"
+
+  # Execute inside of a subshell so that environment changes for ruby-install
+  # don't have a global impact
+  (
+    unset_strict;
+    # Skipping doc installation saves space and time during `docker build`
+    ruby-install --jobs=8 ruby "$ruby_version" -- --disable-install-doc --disable-install-rdoc --disable-install-capi;
```

Comment:
Oh you're right. I noticed `--` used in another case below and I think it's what triggered me.